elliptic-curve: cannot multiply `<C as ProjectiveArithmetic>::ProjectivePoint` by `&NonZeroScalar<C>`

I updated my project to use `k256` v0.7.1, which pulls in `elliptic-curve` v0.8.3. Rebuilding the project throws this error in `elliptic-curve`:

```
error[E0277]: cannot multiply `<C as ProjectiveArithmetic>::ProjectivePoint` by `&NonZeroScalar<C>`
  --> [....]/elliptic-curve-0.8.3/src/ecdh.rs:77:74
   |
77 |     let shared_secret = ProjectivePoint::<C>::from(*public_key.borrow()) * secret_key.borrow();
   |                                                                          ^ no implementation for `<C as ProjectiveArithmetic>::ProjectivePoint * &NonZeroScalar<C>`
   |
   = help: the trait `Mul<&NonZeroScalar<C>>` is not implemented for `<C as ProjectiveArithmetic>::ProjectivePoint`
help: consider further restricting the associated type
   |
75 |     UncompressedPointSize<C>: ArrayLength<u8>, <C as ProjectiveArithmetic>::ProjectivePoint: Mul<&NonZeroScalar<C>>
   |                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

`k256` is built with the following features, if it helps:

```
k256 = { version = "0.7.1", features = ["ecdh", "zeroize", "arithmetic"] }
```

--- a/.github/workflows/elliptic-curve.yml
+++ b/.github/workflows/elliptic-curve.yml
@@ -50,6 +50,7 @@ jobs:
         rust:
           - 1.46.0 # MSRV
           - stable
+          - nightly
     steps:
     - uses: actions/checkout@v1
     - uses: actions-rs/toolchain@v1

--- a/elliptic-curve/src/ecdh.rs
+++ b/elliptic-curve/src/ecdh.rs
@@ -74,7 +74,8 @@ where
     UntaggedPointSize<C>: Add<U1> + ArrayLength<u8>,
     UncompressedPointSize<C>: ArrayLength<u8>,
 {
-    let shared_secret = ProjectivePoint::<C>::from(*public_key.borrow()) * secret_key.borrow();
+    let shared_secret =
+        ProjectivePoint::<C>::from(*public_key.borrow()) * secret_key.borrow().as_ref();
 
     // SharedSecret::new expects an uncompressed point
     // TODO(tarcieri): avoid point encoding when computing shared secret

